vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2022 Vaadin Ltd
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.flow.component.confirmdialog;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.router.RouteConfiguration;
+import com.vaadin.testbench.unit.UIUnitTest;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ConfirmDialogWrapTest extends UIUnitTest {
+
+    ConfirmDialogView view;
+    ConfirmDialogWrap wrap;
+
+    @BeforeEach
+    void init() {
+        RouteConfiguration.forApplicationScope()
+                .setAnnotatedRoute(ConfirmDialogView.class);
+        view = navigate(ConfirmDialogView.class);
+        wrap = wrap(view.dialog);
+    }
+
+    @Test
+    void cancelNotAvailable_throwsException() {
+        wrap.open();
+        assertThrows(IllegalStateException.class, wrap::cancel);
+    }
+
+    @Test
+    void rejectNotAvailable_throwsException() {
+        wrap.open();
+        assertThrows(IllegalStateException.class, wrap::reject);
+    }
+
+    @Test
+    void cancelAvailable_cancelEventClosesDialog() {
+        AtomicInteger close = new AtomicInteger(0);
+        view.dialog.setCancelButton("cancel", event -> close.incrementAndGet());
+        wrap.open();
+
+        wrap.cancel();
+
+        Assertions.assertEquals(1, close.get());
+        Assertions.assertFalse(view.dialog.isOpened());
+    }
+
+    @Test
+    void confirmButton_ClosesDialog() {
```

Comment:
does it make sense to test also that confirmation listener is invoked when present?

Reply:
added